api_end_end_testing/src/test/java/tds/user/api/tests/UserApiTests.java

```diff
@@ -0,0 +1,359 @@
+package tds.user.api.tests;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.jayway.restassured.RestAssured.*;
+import static org.hamcrest.Matchers.*;
+
+import com.jayway.restassured.http.ContentType;
+import org.testng.annotations.Test;
+import tds.user.api.model.UserInfo;
+import tds.user.api.model.RoleAssociation;
+
+/*
+ * Testing User API endpoints
+ */
+public class UserApiTests extends BaseUri{
+
+    private String uriLocation = "/rest/external/user";
+
+    /*
+     * Test of Create User, HTTP POST of /rest/external/user, 201 success item created
+     * Test of Get User, HTTP GET of /rest/external/user/{email}/details, 200 success item found
+     */
+    private UserInfo createUserOneRoleAssoc(String userEmail) {
+
+        List<RoleAssociation> roleAssociations = new ArrayList<RoleAssociation>();
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "44886"));
+
+        UserInfo userInfo = new UserInfo(userEmail, "amy", "watson", "800-332-4747", roleAssociations);
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(userInfo)
+        .when()
+            .post(uriLocation)
+        .then()
+            .statusCode(201)
+            .header("location", endsWith(uriLocation + "/" + userEmail + "/details"));
+
+        // Execute a GET by email to validate that the user was created
+        given()
+            .contentType(ContentType.JSON)
+            .accept(ContentType.JSON)
+            .header(authHeader)
+        .when()
+            .get(uriLocation + "/" + userEmail + "/details")
+        .then()
+            .statusCode(200)
+            .body("firstName", is(userInfo.getFirstName()))
+            .body("lastName", is(userInfo.getLastName()))
+            .body("phoneNumber", is(userInfo.getPhoneNumber()));
```

Comment:
I think we should check the role associations as well to make sure it matches what was passed in.